src/controller/event/handler/replication/replication.go

```diff
@@ -57,13 +60,16 @@ func (r *Handler) IsStateful() bool {
 func (r *Handler) handlePushArtifact(event *event.PushArtifactEvent) error {
 	art := event.Artifact
 	public := false
-	project, err := project.Mgr.Get(art.ProjectID)
-	if err == nil && project != nil {
-		public = project.IsPublic()
-	} else {
-		log.Error(err)
+	project, err := project2.Ctl.Get(context.TODO(), art.ProjectID, project2.Metadata(true))
```

Comment:
`context.TODO()` does not work here because it `orm` instance is missing in `context.TODO()` please use `orm.Context()` in the src/lib pkg.

Reply:
done